azurelinuxagent/ga/exthandlers.py

```diff
@@ -873,42 +875,96 @@ def initialize(self):
+                            handler_configuration=handler_configuration)
         self.set_handler_state(ExtHandlerState.Enabled)
         self.set_handler_status(status="Ready", message="Plugin enabled")
 
     def disable(self):
         self.set_operation(WALAEventOperation.Disable)
         man = self.load_manifest()
+        handler_configuration = None
+        try:
+            handler_configuration = self.load_handler_configuration()
+        except ExtensionError as e:
+            self.logger.debug('Failed to load resource manifest file: {0}'.format(ustr(e)))
+
         disable_cmd = man.get_disable_command()
         self.logger.info("Disable extension [{0}]".format(disable_cmd))
         self.launch_command(disable_cmd, timeout=900,
-                            extension_error_code=ExtensionErrorCodes.PluginDisableProcessingFailed)
+                            extension_error_code=ExtensionErrorCodes.PluginDisableProcessingFailed,
+                            handler_configuration=handler_configuration)
         self.set_handler_state(ExtHandlerState.Installed)
         self.set_handler_status(status="NotReady", message="Plugin disabled")
 
     def install(self):
+        self.set_operation(WALAEventOperation.Install)
         man = self.load_manifest()
+        handler_configuration = None
+        try:
+            handler_configuration = self.load_handler_configuration()
+        except ExtensionError as e:
+            self.logger.debug('Failed to load resource manifest file: {0}'.format(ustr(e)))
+
         install_cmd = man.get_install_command()
         self.logger.info("Install extension [{0}]".format(install_cmd))
         self.set_operation(WALAEventOperation.Install)
         self.launch_command(install_cmd, timeout=900,
-                            extension_error_code=ExtensionErrorCodes.PluginInstallProcessingFailed)
+                            extension_error_code=ExtensionErrorCodes.PluginInstallProcessingFailed,
+                            handler_configuration=handler_configuration)
         self.set_handler_state(ExtHandlerState.Installed)
 
     def uninstall(self):
         try:
             self.set_operation(WALAEventOperation.UnInstall)
             man = self.load_manifest()
+            handler_configuration = None
+            try:
+                handler_configuration = self.load_handler_configuration()
+            except ExtensionError as e:
+                self.logger.debug('Failed to load resource manifest file: {0}'.format(ustr(e)))
+
             uninstall_cmd = man.get_uninstall_command()
             self.logger.info("Uninstall extension [{0}]".format(uninstall_cmd))
-            self.launch_command(uninstall_cmd)
+            self.launch_command(uninstall_cmd,
+                                handler_configuration=handler_configuration)
         except ExtensionError as e:
             self.report_event(message=ustr(e), is_success=False)
 
+    def update(self, version=None):
```

Comment:
there is another update @ 986

Reply:
Done. Sorry, was keeping the ordering correct.